service/controller/resource/appmigration/create.go

```diff
@@ -68,35 +68,60 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
 	}
 
+	ch := make(chan response)
+
 	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
 	defer cancel()
 
-	chartConfigs, err := cc.Client.TenantCluster.G8s.CoreV1alpha1().ChartConfigs("giantswarm").List(listOptions)
-	if tenant.IsAPINotAvailable(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available yet")
+	go func() {
+		chartConfigs, err := cc.Client.TenantCluster.G8s.CoreV1alpha1().ChartConfigs("giantswarm").List(listOptions)
```

Comment:
I wonder why we use context timeouts here when this line doesn't take `ctx` as an arg. Why not `time.After`? We can also avoid `response` type. E.g.:

```go
	ch := make(chan struct{})
 	var chartConfigList ChartConfigList
	var err error

	go func() {
		chartConfigs, err = cc.Client.TenantCluster.G8s.CoreV1alpha1().ChartConfigs("giantswarm").List(listOptions)
 		close(ch)
	}

	select {
	case <-ch:
		// Fall through.
	case <-time.After(3*time.Second):
		// Set status so we don't try to connect to the tenant cluster
		// again in this reconciliation loop.
		cc.Status.TenantCluster.IsUnavailable = true

		r.logger.LogCtx(ctx, "level", "debug", "message", "timeout getting chartconfig crs")
		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
		return nil
	}
```

Reply:
I searched around this topic, and found this. 
```
So a Context may be used to signal cancelation to arbitrary number of goroutines and places. A Timer can only be used to signal one target, the one who receives the value from its channel. 
...
Also if you're using / working with code that already supports / expects context.Context, then it ins't a question which one to use.
```

https://stackoverflow.com/questions/42359053/how-to-decide-between-a-context-withdeadline-or-a-simple-timer

I think we can use `context` for a while.

Reply:
Yes the current code works but I think you're right using time.After is cleaner

I need to think about it some more. I want to use an identical approach in both app-operator and cluster-operator. As we have the same problems with both.

Reply:
I mean if `CoreV1alpha1().ChartConfigs("giantswarm").List(listOptions)` was taking context and handling timeouts properly we wouldn't have to deal with that `select`. So I think `time.After` in 100% legit here.

Reply:
Now implemented and much cleaner. Thx! <3

I'll do the same in app-operator.